### ISSUE TITLE
A file cannot be read and deleted by the same query reliably

### ISSUE THREAD

**Issue by fcavalieri** [28msec/zorba#135]
When a file is read using the file module a streamable string/binary64 item is created. The stream is not eagerly read. The stream is closed (destroyed) only after the query ends (not even when the variable goes out of scope). If the same query tries to delete the same file, on windows an error occurs (the file is open, thus it cannot be deleted). On linux no error occurs but there is something wrong, as the string is never materialized, and the file seems to be deleted. Timing? Not really deleted?

module namespace fct = "http://www.zorba-xquery.com/tests/file_copy_test";
import module namespace file = "http://expath.org/ns/file";
declare namespace ann = "http://zorba.io/annotations";
declare %ann:nondeterministic %ann:sequential
function fct:test-copy($rbktPath as xs:string, $file as xs:string)
{
  variable $fileSrc := fn:concat($rbktPath, "/Queries/zorba/file/copy_files/", $file);
  variable $fileDest := fn:concat($rbktPath, "/Queries/zorba/file/copy_files/", $file, ".out");

  (: make sure the output file does not exist :)
  if (file:exists($fileDest)) then
    exit returning "The output file already exists. This is not good!";
  else
    ();

  (: copy the file :)
  file:copy($fileSrc, $fileDest);

  variable $result := true();
  fct:compare-binary($fileSrc, $fileDest);
  (: delete the file before we terminate :)
  file:delete($fileDest);
    
  if (file:exists($fileDest)) then
    "The output file must not be here anymore. This is not good!"
  else
    $result  
};

declare %ann:nondeterministic %ann:cache function fct:compare-binary($file1 as xs:string, $file2 as xs:string)
{
  file:read-binary($file1) eq file:read-binary($file2)
};
